lib/nerves_hub/devices.ex

```diff
@@ -698,20 +698,22 @@ defmodule NervesHub.Devices do
 
   @spec update_deployment(Device.t(), Deployment.t()) :: Device.t()
   def update_deployment(device, deployment) do
-    device
-    |> Ecto.Changeset.change()
-    |> Ecto.Changeset.put_change(:deployment_id, deployment.id)
-    |> Repo.update!()
-    |> Repo.preload(:deployment, force: true)
+    %Device{} =
+      device
+      |> Ecto.Changeset.change()
+      |> Ecto.Changeset.put_change(:deployment_id, deployment.id)
+      |> Repo.update!()
+      |> Repo.preload(:deployment, force: true)
   end
 
   @spec clear_deployment(Device.t()) :: Device.t()
   def clear_deployment(device) do
-    device
-    |> Ecto.Changeset.change()
-    |> Ecto.Changeset.put_change(:deployment_id, nil)
-    |> Repo.update!()
-    |> Repo.preload(:deployment, force: true)
+    %Device{} =
+      device
+      |> Ecto.Changeset.change()
+      |> Ecto.Changeset.put_change(:deployment_id, nil)
+      |> Repo.update!()
+      |> Repo.preload(:deployment, force: true)
```

Comment:
Likewise here. Why do another DB action when we know nothing will load. We can simply do the work and skip the call _and_ it should fix dialyzer
```suggestion
      device
      |> Ecto.Changeset.change()
      |> Ecto.Changeset.put_change(:deployment_id, nil)
      |> Repo.update!()
      |> Map.put(:deployment, nil)
```